packages/itwin/tree-widget/src/components/trees/VisibilityTreeRenderer.tsx

```diff
@@ -14,13 +14,44 @@ import type { NodeCheckboxRenderProps } from "@itwin/core-react";
 import type { IPresentationTreeDataProvider } from "@itwin/presentation-components";
 import type { VisibilityTreeFilterInfo } from "./Common";
 
+/**
+ * Props for visibility tree node renderer.
+ * @public
+ */
+export interface VisibilityTreeRendererProps {
+  /**
+   * Specifies whether the icon at the left of the node label should be rendered.
+   */
+  iconsEnabled: boolean;
+  /**
+   * Specifies whether node description should be enabled.
+   */
+  descriptionEnabled: boolean;
+  /**
+   * Defines the offset in pixels of how much each hierarchy level should be offset to the right from the checkbox.
+   * Defaults to `20`.
+   */
+  levelOffset?: number;
+  /**
+   * Defines the size in pixels of the expansion toggle. It is used to keep same hierarchy nodes with children and nodes without children in the same line.
+   * @note This value applies only to the leaf nodes.
+   * Defaults to `24`.
+   */
+  expansionToggleWidth?: number;
+  /**
+   * Specifies whether the root node be expanded at all times.
+   * Defaults to `false`.
+   */
+  disableRootNodeCollapse?: boolean;
+}
+
 /**
  * Creates Visibility tree renderer which renders nodes with eye checkbox.
  * @public
  */
-export const useVisibilityTreeRenderer = (iconsEnabled: boolean, descriptionsEnabled: boolean) => {
+export const useVisibilityTreeRenderer = (visibilityTreeRendererProps: VisibilityTreeRendererProps) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  const nodeRenderer = useCallback(createVisibilityTreeNodeRenderer(iconsEnabled, descriptionsEnabled), [iconsEnabled, descriptionsEnabled]);
+  const nodeRenderer = useCallback(createVisibilityTreeNodeRenderer(visibilityTreeRendererProps), [visibilityTreeRendererProps]);
```

Comment:
I think there is no need to use `useCallback` in this method. At least with current implementation because it will create new callback every render unless `visibilityTreeRendererProps` have stable reference. However, looking at how it is used in Categories and Models trees it does not have stable reference.

Also, if `useCallback`s are removed from this method I think it should be renamed to `createVisibilityTreeRenderer` as it does not use any hooks.

Reply:
fixed